cmd/jaeger/internal/integration/e2e_integration.go

```diff
@@ -92,14 +93,18 @@ func (s *E2EStorageIntegration) e2eInitialize(t *testing.T, storage string) {
 	}
 	cmd.Start(t)
 
-	s.SpanWriter, err = createSpanWriter(logger, otlpPort)
+	spanWriter, err := createSpanWriter(logger, otlpPort)
 	require.NoError(t, err)
+	s.TraceWriter = v1adapter.NewTraceWriter(spanWriter)
```

Comment:
If you look at span_writer.go you will notice that it is already using ptrace model to actually communicate to the collector. So rather than using the adapter, I would simply change the signature of this span writer to accept ptrace directly (and rename it to trace_writer.go)